src/DynamoPackagesUI/Utilities/PackageManagerCommands.cs

```diff
@@ -0,0 +1,485 @@
+                p.ID = DownloadRequest.asset_id;
+                Application.Current.Dispatcher.Invoke((Action)(() =>
+                {
+                    dynamoViewModel.Model.GetPackageManagerExtension().PackageLoader.Load(p);
+                }));
+                packageDownloadHandle.DownloadState = PackageDownloadHandle.State.Installed;
+                this.PackageInstallPath = string.Empty;
+            }
+        }
+
+        /// <summary>
+        /// Download and Install Dynamo Package
+        /// </summary>
+        /// <param name="pkg"></param>
+        public void DownloadAndInstall(string pkg)
+        {
+            string[] temp = pkg.Split(',');
+            PackageManagerRequest req = new PackageManagerRequest("assets/" + temp[0], Method.GET);
+            ResponseWithContentBody<dynamic> response = DPClient.ExecuteAndDeserializeDynamoRequest(req);
+            DownloadRequest = response.content;
+
+            PackageManagerRequest fileReq = new PackageManagerRequest(@"files/download?file_ids=" + temp[1] + "&asset_id=" + temp[0], Method.GET, true);
+            Response res = DPClient.ExecuteDynamoRequest(fileReq);
+            var pathToPackage = DPClient.GetFileFromResponse(res);
+            InstallPackage(pathToPackage);
+        }
+
+        public string GetCustomPathForInstall()
+        {
+            string downloadPath = string.Empty;
+
+            downloadPath = GetDownloadPath();
+
+            if (String.IsNullOrEmpty(downloadPath))
+                return string.Empty;
+
+            PackageInstallPath = downloadPath;
+
+            return downloadPath;
+        }
+
+        private string GetDownloadPath()
+        {
+            var args = new PackagePathEventArgs();
+
+            ShowFileDialog(args);
+
+            if (args.Cancel)
+                return string.Empty;
+
+            return args.Path;
+        }
+
+        private void ShowFileDialog(PackagePathEventArgs e)
+        {
+            string initialPath = dynamoViewModel.Model.PathManager.DefaultPackagesDirectory;
+
+            e.Cancel = true;
+
+            var dialog = new DynamoFolderBrowserDialog
+            {
+                // Navigate to initial folder.
+                SelectedPath = initialPath,
+                Title = "Install Path",
+                Owner = this.ParentWindow
+            };
+            Application.Current.Dispatcher.Invoke((Action)(() =>
+            {
+                if (dialog.ShowDialog() == System.Windows.Forms.DialogResult.OK)
+                {
+                    e.Cancel = false;
+                    e.Path = dialog.SelectedPath;
+                }
+            }));
+        }
+
+        /// <summary>
+        /// Install button click event
+        /// </summary>
+        /// <param name="asset"></param>
+        /// <param name="version"></param>
+        /// <returns></returns>
+        public string PackageOnExecuted(dynamic asset, dynamic version)
```

Comment:
This looks like a callback method, why is this method public? Should this be named as `OnInstallPackage`?